src/main/scala/com/github/sbt/sbom/SbtUpdateReport.scala

```diff
@@ -62,7 +63,31 @@ object SbtUpdateReport {
       GraphModuleId(sbtId.organization, sbtId.name, sbtId.revision)
   }
 
-  def fromConfigurationReport(report: ConfigurationReport, rootInfo: ModuleID): ModuleGraph = {
+
+  def getModuleQualifier(moduleReport: ModuleReport, log: Option[Logger] = None): Map[String, String] = {
+    val qualifier = new mutable.HashMap[String, String]()
+
+    // Getting artifact with the same name as module name as purl qualifier
+    val moduleArtifacts = moduleReport.artifacts.filter(ar =>{
+      ar._1.name.equals(moduleReport.module.name)
+    })
+
+    moduleArtifacts.size match {
+      case 0 => () // ignore empty found artifacts
+      case x =>
+        if (x > 1 && log.isDefined) {
+          log.foreach(_.warn("Multiple artifacts with the same name as module name are detected. Taking the first artifact match as Purl qualifier."))
```

Comment:
is this meant to avoid #152? did you see cases where including the qualifier was not sufficient?